rows_test.go

```diff
@@ -0,0 +1,327 @@
+		`INSERT INTO TestAtomicTypes (key, testString, testBytes, testInt, testFloat, testBool) VALUES
+		("general", "hello", CAST ("hello" as bytes), 42, 42, TRUE), 
+		("negative", "hello", CAST ("hello" as bytes), -42, -42, TRUE),
+		("maxint", "hello", CAST ("hello" as bytes), 9223372036854775807 , 42, TRUE),
+		("minint", "hello", CAST ("hello" as bytes), -9223372036854775808 , 42, TRUE),
+		("nan", "hello" ,CAST("hello" as bytes), 42, CAST("nan" AS FLOAT64), TRUE), 
+		("pinf", "hello" ,CAST("hello" as bytes), 42, CAST("inf" AS FLOAT64), TRUE),
+		("ninf", "hello" ,CAST("hello" as bytes), 42, CAST("-inf" AS FLOAT64), TRUE), 
+		("nullstring", null, CAST ("hello" as bytes), 42, 42, TRUE), 
+		("nullbytes", "hello", null, 42, 42, TRUE),
+		("nullint", "hello", CAST ("hello" as bytes), null, 42, TRUE), 
+		("nullfloat", "hello", CAST ("hello" as bytes), 42, null, TRUE), 
+		("nullbool", "hello", CAST ("hello" as bytes), 42, 42, null)
+		`)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	type testAtomicTypesRow struct {
+		key        string
+		testString string
+		testBytes  []byte
+		testInt    int
+		testFloat  float64
+		testBool   bool
+	}
+
+	type wantError struct {
+		scan  bool
+		close bool
+		query bool
+	}
+
+	tests := []struct {
+		name      string
+		input     string
+		want      []testAtomicTypesRow
+		wantError wantError
+	}{
+
+		{
+			name:  "general read",
+			input: `SELECT * FROM TestAtomicTypes WHERE key = "general"`,
+			want: []testAtomicTypesRow{
+				{key: "general", testString: "hello", testBytes: []byte("hello"), testInt: 42, testFloat: 42, testBool: true},
+			},
+		},
+		{
+			name:  "negative int and float",
+			input: `SELECT * FROM TestAtomicTypes WHERE key = "negative"`,
+			want: []testAtomicTypesRow{
+				{key: "negative", testString: "hello", testBytes: []byte("hello"), testInt: -42, testFloat: -42, testBool: true},
+			},
+		},
+		{
+			name:  "max int",
+			input: `SELECT * FROM TestAtomicTypes WHERE key = "maxint"`,
+			want: []testAtomicTypesRow{
+				{key: "maxint", testString: "hello", testBytes: []byte("hello"), testInt: 9223372036854775807, testFloat: 42, testBool: true},
+			},
+		},
+		{
+			name:  "min int",
+			input: `SELECT * FROM TestAtomicTypes WHERE key = "minint"`,
+			want: []testAtomicTypesRow{
+				{key: "minint", testString: "hello", testBytes: []byte("hello"), testInt: -9223372036854775808, testFloat: 42, testBool: true},
+			},
+		},
+		{
+			name:  "special float positive infinity",
+			input: `SELECT * FROM TestAtomicTypes WHERE key = "pinf"`,
+			want: []testAtomicTypesRow{
+				{key: "pinf", testString: "hello", testBytes: []byte("hello"), testInt: 42, testFloat: math.Inf(1), testBool: true},
+			},
+		},
+		{
+			name:  "special float negative infinity",
+			input: `SELECT * FROM TestAtomicTypes WHERE key = "ninf"`,
+			want: []testAtomicTypesRow{
+				{key: "ninf", testString: "hello", testBytes: []byte("hello"), testInt: 42, testFloat: math.Inf(-1), testBool: true},
+			},
+		},
+		/*{
```

Comment:
Please remove if we won't be adding this one.

Reply:
removed